### PR TITLE
Fix FIO service monitor and add e2e test

### PR DESCRIPTION
**Pull request by Vincent056**
This pr adds e2e test to test the servicemonitor is working correctly, we will check and make sure promethus is able to scrap the FIO metric endpoint.

### CODE REVIEW

File: tests/e2e/helpers.go [hunk 1]
```diff
@@ -2306,3 +2306,88 @@ func parseMetric(t *testing.T, content, metric string) int {
...
+		t.Fatalf("No valid JSON found in output")
+	}
+
+	t.Logf("metrics output:\n%s\n", out[0:100])
+	var responseData struct {
+		Data struct {
+			ActiveTargets []Metric `json:"activeTargets"`
+		} `json:"data"`
+	}
+	err := json.Unmarshal([]byte(out), &responseData)
```
Comment (rhmdnd) [inline tests/e2e/helpers.go:2357]: Looks like this choked on the Prometheus output. I wonder if we need to update how parseMetric is handling it.

File: tests/e2e/e2e_test.go [hunk 2]
```diff
@@ -165,6 +165,19 @@ func TestMetricsHTTPVersion(t *testing.T) {
...
+	testName := testIntegrityNamePrefix + "-metrics-target-service-monitoring"
+	setupFileIntegrity(t, f, testctx, testName, namespace, nodeWorkerRoleLabelKey, defaultTestGracePeriod)
+	defer testctx.Cleanup()
+	// Create service account
+	createServiceAccount(t, namespace)
+	defer deleteServiceAccount(t, namespace) // Ensure the service account is deleted after the test
+
+	metricsTargets := getPrometheusMetricTargets(t, namespace)
+
+	assertServiceMonitoringMetricsTarget(t, metricsTargets, 2)
```
Comment (rhmdnd) [inline tests/e2e/e2e_test.go:179]: Looks like this is catching a legitimate failure:

```
helpers.go:2298: metrics output:
{"status":"success","data":{"activeTargets":[{"discoveredLabels":{"__address__":"10.128.0.32:8443","__meta_kubernetes_endpoint_address_target_kind":"Pod","__meta_kubernetes_endpoint_address_target_name":"openshift-apiserver-operator-76dd58979b-46tmh","__meta_kubernetes_endpoint_node_name":"ip-10-0-9 
```

File: tests/e2e/helpers.go [hunk 3]
```diff
@@ -2262,3 +2288,97 @@ func parseMetric(t *testing.T, content, metric string) int {
...
+	}
+
+	if err := runOCandCheckError(t, []string{
+		"adm", "policy", "add-cluster-role-to-user", "cluster-monitoring-view", "-z", PromethusTestSA, "-n", operatorNamespace}); err != nil {
+		t.Fatalf("Failed to add cluster role to user: %v", err)
+	}
+}
+
+// CleanupRBACForMetricsTest deletes the service account
+func CleanupRBACForMetricsTest(t *testing.T, operatorNamespace string) {
```
Comment (yuumasato) [inline tests/e2e/helpers.go:2306]: Should we also delete the cluster role?

Comment (Vincent056) [reply tests/e2e/helpers.go:2306]: I think we didn't create that role

**Comment by yuumasato** [comment]
@Vincent056 we need a rebase, some of the fixes merged for CI led to conflicts.

**Review by yuumasato — APPROVED** [review]
/approve

**Comment by yuumasato** [comment]
@xiaojiey @BhargaviGudi please check this PR

**Comment by xiaojiey** [comment]
Verification pass with index image built from https://github.com/openshift/file-integrity-operator/pull/545, with both 4.18 and 4.15 payload.
```
% token=`oc create token prometheus-k8s -n openshift-monitoring`
% oc -n openshift-monitoring exec -c prometheus prometheus-k8s-0 -- curl -k -H "Authorization: Bearer $token" https://prometheus-k8s.openshift-monitoring.svc.cluster.local:9091/api/v1/targets | jq -r  > target.json
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 9347k    0 9347k    0     0  7201k      0 --:--:--  0:00:01 --:--:-- 7201k
% cat target.json| jq '.data.activeTargets[]|select(.labels.namespace == "openshift-file-integrity") .health'
"up"
"up"
```

**Review by yuumasato — APPROVED** [review]
/approve

**Pull request merged**